make/Makefile.molecule.mk

```diff
@@ -155,18 +143,14 @@ else
 endif
 
 .prepare-molecule-data-volume:
-	$(DORP) volume create molecule-tests-volume
-	$(DORP) create -v molecule-tests-volume:/data --name molecule-volume-helper docker.io/busybox true
-	$(DORP) cp "${HELM_CHARTS_REPO}" molecule-volume-helper:/data/helm-charts-repo
-	$(DORP) cp "${ROOTDIR}/operator/" molecule-volume-helper:/data/operator
-	$(DORP) cp "${MOLECULE_KUBECONFIG}" molecule-volume-helper:/data/kubeconfig
-	$(DORP) rm molecule-volume-helper
+	podman volume create molecule-tests-volume
+	podman create -v molecule-tests-volume:/data --name molecule-volume-helper docker.io/busybox true
+	podman cp "${HELM_CHARTS_REPO}" molecule-volume-helper:/data/helm-charts-repo
+	podman cp "${ROOTDIR}/operator/" molecule-volume-helper:/data/operator
+	podman cp "${MOLECULE_KUBECONFIG}" molecule-volume-helper:/data/kubeconfig
+	podman rm molecule-volume-helper
 
 ## molecule-test: Runs Molecule tests using the Molecule docker image
 molecule-test: .ensure-operator-repo-exists .ensure-operator-helm-chart-exists .prepare-add-host-args molecule-build .prepare-molecule-data-volume .create-operator-pull-secret
-ifeq ($(DORP),docker)
```

Comment:
are we sure we can't run these using docker? I used to have only docker at one point and this all worked. This was during a time we used the old , now deleted, molecule base images so perhaps that is the difference?

Reply:
Yeah this was part of the fallout from those old images getting deleted. We _have_ to run the tests with podman now since the ansible/molecule tests no longer support docker.

Reply:
When you try to run with docker you get:
![Screenshot from 2022-06-23 14-20-03](https://user-images.githubusercontent.com/6226732/175367852-92fc85ce-19a3-43da-864b-451592b950cd.png)

from: https://github.com/kiali/kiali/runs/7017770872?check_suite_focus=true

Reply:
OK, I'm looking at all of this a little closer, and I think you can leave the Makefile.molecule.mk file alone. I run the full molecule test suite fine on a machine that ONLY has podman (see the test results from last night: https://github.com/jmazzitelli/kiali-molecule-test-logs/blob/openshift/molecule-tests-2022-06-23_02-00-02/results.log)

So there is no need to remove DORP from the Molecule makefile.

It seems the bulk of the changes are to support KinD. So make those changes (you can require "docker" for the Kind specific things if need be). But don't require the machine to have docker since there is no need to for some use cases (i.e. mine :). Having just podman works fine in my case.

Reply:
conversely, if you NEED podman and you don't have it, you can fail at that point.

Reply:
> So there is no need to remove DORP from the Molecule makefile.

The main motivation was to make it clear that you can't pass `DORP=docker` and expect the molecule tests to pass. We could keep the `DORP` option and just make it so that the molecule tests always run podman even when passing `DORP=docker` but imo that's confusing.

> If you NEED docker and it is missing, then you can fail as is the case for the Kind stuff like:

If you are running the molecule tests in kind (which is how CI runs them), then you need both docker and podman. Because of this, the changes here were two fold: 1. Ensure molecule tests are always run with podman 2. Ensure kind is always started with docker (since podman support is experimental). If you're running the molecule tests against minikube or openshift then you'd only need podman.

> finally, leave the DORP , --dorp options in place - don't delete them. I have a feeling in the future the molecule peeps will add back docker support and we can just re-use it.

I don't think they will :). If you're going to go through the trouble of removing support and breaking people that way, I'd be surprised if they added it back after ripping off the bandaid. But if we're going to leave the DORP options, what should be the behavior then? Either we can ignore `DORP` even when `DORP=docker` is passed or allow `DORP=docker` but then the molecule tests will always fail with that option. The same is true of the kind script since that should always run with docker and will need to ignore `DORP=podman` or fail. In the future we could support rootless podman with kind. I got this to work locally but I had to tweak some kernel settings to do it which doesn't give me confidence in this working for CI.

Reply:
From what I see then, it seems only KinD needs this special environment that requires both podman and docker, correct? (That has to be - because in my case, I don't even have docker and all works well - I am not using KinD). So for the KinD specific script and make code, hardcode docker or podman where it makes sense.

For everything else, use DORP. If DORP must be "podman" (like in the "molecule-run" make target) then simply fail fast.
So...

```
ifeq($(DORP),docker)
   echo "Must use podman" && exit 1
endif
```

```bash
[ "$DORP" == "docker" ] && echo "Must use podman" && exit 1
```

I do NOT need docker and things work for me. I want to keep it that way. And I don't really want to have to remember all the places where I define DORP env var and where I pass -dorp to the scripts in on my different machines to now change it (becuse now in this PR the `-dorp` option is unknown/invalid). I want to keep what I have. Not to mention people out in the community using these things - we don't want to break them (I know there is at least one other person out in the community running these molecule tests).

Reply:
> From what I see then, it seems only KinD needs this special environment that requires both podman and docker, correct? (That has to be - because in my case, I don't even have docker and all works well - I am not using KinD). So for the KinD specific script and make code, hardcode docker or podman where it makes sense.

Yes but I just got rootless podman working with kind on my local environment so you could run kind with either docker or podman and both should work provided you can already run rootless podman on your machine. CI will use docker for kind and podman for the molecule tests because in order to get rootless podman working, you may need to modify some kernel settings and I have strong doubts about getting that working in CI.

> I do NOT need docker and things work for me. I want to keep it that way. And I don't really want to have to remember all the places where I define DORP env var and where I pass -dorp to the scripts in on my different machines to now change it (becuse now in this PR the -dorp option is unknown/invalid). I want to keep what I have. Not to mention people out in the community using these things - we don't want to break them (I know there is at least one other person out in the community running these molecule tests).

That's fine I'll add back in the dorp opt and just have the script fail if you pass docker as you suggested. If other folks out there are passing `dorp=docker` to the molecule tests they are already broken.

Reply:
@jmazzitelli I added back all the dorp settings but now I can no longer run things locally for some reason so I'm not sure if this will work or not in CI. The molecule tests are complaining about the helm chart not being built. I've got no clue why since I can see the helm chart being built in the correct location and it's very difficult to track settings being plumbed through 3 different scripts and makefiles.

Reply:
If the nightly molecule run passes then that's good enough for me personally.